File: prime.rb
puts "Enter a number:"
num = gets.to_i

is_prime = true

if num <= 1
  is_prime = false
else
  (2..Math.sqrt(num)).each do |i|
    if num % i == 0
      is_prime = false
      break
    end
  end
end

if is_prime
  puts "#{num} is a Prime Number"
else
  puts "#{num} is Not a Prime Number"
end

File: converted-code/prime.ts
typescript
const readline = require('readline').createInterface({
  input: process.stdin,
  output: process.stdout
});

readline.question('Enter a number: ', (num) => {
  const numValue = parseInt(num);

  let isPrime = true;

  if (numValue <= 1) {
    isPrime = false;
  } else {
    for (let i = 2; i <= Math.sqrt(numValue); i++) {
      if (numValue % i === 0) {
        isPrime = false;
        break;
      }
    }
  }

  if (isPrime) {
    console.log(`${numValue} is a Prime Number`);
  } else {
    console.log(`${numValue} is Not a Prime Number`);
  }

  readline.close();
});